context/fa2.ts

```diff
@@ -6,18 +6,26 @@ type makeFa2MichelsonParam = {
   fa2Address: string;
 };
 
-export const makeFa2Michelson = ({
-  walletAddress,
-  targetAddress,
-  tokenId,
-  amount,
-  fa2Address,
-}: makeFa2MichelsonParam) => `{
+export const makeFa2Michelson = (params: makeFa2MichelsonParam[]) => {
+  if (params.length === 0) throw new Error("Empty fa2 params");
+
+  return `{
     DROP;
-    PUSH address "${fa2Address}";
+    PUSH address "${params[0].fa2Address}";
     CONTRACT %transfer (list (pair (address %from_) (list %txs (pair (address %to_) (pair (nat %token_id) (nat %amount))))));
     IF_NONE { PUSH string "contract dosen't exist" ; FAILWITH } { } ;
     PUSH mutez 0 ;
-    PUSH (list (pair (address %from_) (list %txs (pair (address %to_) (pair (nat %token_id) (nat %amount)))))) {Pair "${walletAddress}" {Pair "${targetAddress}" (Pair ${tokenId} ${amount})}} ;
+    PUSH (list (pair (address %from_) (list %txs (pair (address %to_) (pair (nat %token_id) (nat %amount)))))) {Pair "${
+      params[0].walletAddress
+    }" 
```

Comment:
```suggestion
    }"  {
```
sorry :pray:  , I made wrong suggestion again ... that was only work for one. this time should be right.